pyogrio/geopandas.py

```diff
@@ -190,10 +195,21 @@ def write_dataframe(df, path, layer=None, driver=None, encoding=None, **kwargs):
     # TODO: may need to fill in pd.NA, etc
     field_data = [df[f].values for f in fields]
 
+    geometry_type = None
     if not df.empty:
         # TODO: validate geometry types, not all combinations are valid
-        geometry_type = geometry.type.unique()[0]
-    else:
+        geometry_types = geometry.type.unique()
+        if len(geometry_types) == 1:
+            geometry_type = geometry_types[0]
+        elif len(geometry_types == 2):
+            if "Polygon" in geometry_types and "MultiPolygon" in geometry_types:
+                geometry_type = "MultiPolygon"
+            elif "LineString" in geometry_types and "MultiLineString" in geometry_types:
+                geometry_type = "MultiLineString"
+            elif "Point" in geometry_types and "MultiPoint" in geometry_types:
+                geometry_type = "MultiPoint"
```

Comment:
Should this part only be done if `promote_to_multi=True` ? 
(and otherwise leave a mixture as Unknown)

Reply:
I'll have to look into this some more. 

For Geopackage your are certainly right. But eg. shapefile doesn't know/make the difference between multipolygon and polygon, so there the current behaviour is probably better.

Reply:
So that we can deal with proper handling of "Unknown" in a separate PR, how about this?

```Python
geometry_type = "Unknown"

if not df.empty:
    geometry_types = geometry.type.unique()
    if promote_to_multi:
        ...what you have now...

    else:
        # default to type of first geometry
        geometry_type = geometry_types[0]
```